types/mydecimal.go

```diff
@@ -2010,7 +2010,9 @@ func doDivMod(from1, from2, to, mod *MyDecimal, fracIncr int) error {
 	}
 	if prec1 <= 0 {
 		/* short-circuit everything: from1 == 0 */
+		resultFrac := to.resultFrac
 		*to = zeroMyDecimal
+		to.resultFrac = resultFrac
```

Comment:
Please add a comment here to explain these three lines.
And it's better to add a unit test in mydecimal_test.go

Reply:
> Hi, after seeing your pull request, it may have other problems with `zeroMyDecimal` like `SELECT 1.111-1.111;`
> 
> Could you take a review and fix them too?

OK

Reply:
```
	for _, tc := range testCases {
		sig, err := funcs[ast.Mod].getFunction(s.ctx, s.datumsToConstants(types.MakeDatums(tc.args...)))
		c.Assert(err, IsNil)
		c.Assert(sig, NotNil)
		val, err := evalBuiltinFunc(sig, chunk.Row{})
		c.Assert(err, IsNil)
		c.Assert(val, testutil.DatumEquals, types.NewDatum(tc.expect))
	}
```
In TestArithmeticMod, `c.Assert` uses `DatumEquals`,  can it distinguish 0 from 0.0000...?